frontend/src/components/create-contract/PaymentSummary.tsx

```diff
@@ -1,11 +1,29 @@
-const PaymentSummary = () => {
+import { CreateFormValues } from 'components/forms/types';
+
+//
+const PaymentSummary = ({
+  tokenName,
+  data,
+}: {
+  tokenName?: string;
+  data: CreateFormValues;
```

Comment:
You can actually avoid the prop here!

Inside the body of this component, use `useFormikContext`. Destructure that into `values` (and lots of other stuff is available.)

Bonus points if we can wrap that in our own `useForm()` hook, to keep the components as readable as possible.

https://formik.org/docs/api/useFormikContext